packages/commerce-sdk-react/src/hooks/ShopperBaskets/cache.ts

```diff
@@ -109,76 +54,430 @@ const invalidateCustomerBasketsQuery = (
+                          }
+                      ]
+                    : [])
+            ]
+        }
+    },
+    addItemToBasket(customerId, {parameters}, response) {
+        return {
+            update: [
+                {queryKey: getBasket.queryKey(parameters)},
+                ...(customerId
+                    ? [
+                          {
+                              queryKey: getCustomerBaskets.queryKey({...parameters, customerId}),
+                              updater: (oldData: CustomerBasketsResult | undefined) =>
+                                  customerBasketsUpdater(parameters, response, oldData)
+                          }
+                      ]
+                    : [])
+            ]
+        }
+    },
+    addPaymentInstrumentToBasket(customerId, {parameters}, response) {
+        return {
+            update: [
+                {queryKey: getBasket.queryKey(parameters)},
+                ...(customerId
+                    ? [
+                          {
+                              queryKey: getCustomerBaskets.queryKey({...parameters, customerId}),
+                              updater: (oldData: CustomerBasketsResult | undefined) =>
+                                  customerBasketsUpdater(parameters, response, oldData)
+                          }
+                      ]
+                    : [])
+            ]
+        }
+    },
+    addPriceBooksToBasket(customerId, {parameters}) {
         // TODO: Convert invalidate to an update that removes the matching basket
-        ...invalidateCustomerBasketsQuery(customerId, parameters),
-        remove: [
-            and(
-                matchParameters(toGetBasketParameters(parameters)),
-                matchesPath(getBasketPath(parameters))
-            )
-        ]
-    }),
-    mergeBasket: updateBasketWithResponseBasketId,
-    removeCouponFromBasket: updateBasket,
-    removeGiftCertificateItemFromBasket: TODO('removeGiftCertificateItemFromBasket'),
-    removeItemFromBasket: updateBasket,
-    removePaymentInstrumentFromBasket: updateBasket,
-    removeShipmentFromBasket: TODO('removeShipmentFromBasket'),
-    transferBasket: TODO('transferBasket'),
-    updateBasket: updateBasket,
-    updateBillingAddressForBasket: updateBasket,
-    updateCustomerForBasket: updateBasket,
-    updateGiftCertificateItemInBasket: TODO('updateGiftCertificateItemInBasket'),
-    updateItemInBasket: updateBasket,
-    updatePaymentInstrumentInBasket: updateBasket,
-    updateShipmentForBasket: TODO('updateShipmentForBasket'),
-    updateShippingAddressForShipment: updateBasket,
-    updateShippingMethodForShipment: updateBasket
+        const invalidations = invalidateCustomerBasketsQuery(customerId, parameters)
+        invalidations?.invalidate?.push({queryKey: getPriceBooksForBasket.queryKey(parameters)})
+        return {
+            ...invalidations,
+            update: [{queryKey: getBasket.queryKey(parameters)}]
```

Comment:
should update come before invalidations? Does order matter?

Reply:
The order we define the cache updates here doesn't matter, since we're working with an object were the concept of order really doesn't exist.

The `updateCache` method defines in which order the cache updates are executed:
https://github.com/SalesforceCommerceCloud/pwa-kit/blob/5c604835e11b41ca55418f55b2e5aaa4b28cd7a5/packages/commerce-sdk-react/src/hooks/utils.ts#L18-L29

Reply:
The order doesn't matter, it's just merging different properties into the result object (which will always look like `{update: [], invalidate: [], remove: []}`).